README.md

```diff
@@ -357,6 +357,46 @@ TrySafe::fromCallable(fn() => $getTheUltimateAnswerOrThrow(true))
  
 _Disclaimer:_ As you probably noticed, we reflect possible exception in return type now, but on the other side, we've lost the information that wrapped success value is `float`.  This applies also to `Option`, `ArrayList` etc. Unfortunately there is no silver bullet solution, until PHP have the generics implemented (but hey, have look at [phpstan generics templates](https://medium.com/@ondrejmirtes/generics-in-php-using-phpdocs-14e7301953)). 
 
+#### TrySafe recovery
+
+We have already learned, that `TrySafe` can server to chaining dependant operations that can fail (via `flatMap`). How about if we want to have some fallback / recovery in the middle of that chain?
+
+This is where `recover*` methods come in handy. Let's take a look on this example:
+
+```php
+/** @var callable(Throwable): TrySage<Gps> */
+$recovery = fn (Throwable $ex): TrySafe => $backupApi->findGps($query);
+
+/** @var TrySafe<int> */
+$distance = $api
+    ->findGps($query)
+    ->recoverWith($findGpsFromBackupApi)
+    ->map(fn (Gps $gps) => $this->getDistance($home));
+```
+
+There is three `recover*` methods:
+- `recover` - return directly value to which you want to recover
```

Comment:
Return value, you want computation to be recovered with

Reply:
End the sentence with **with** might be confusing with the line bellow, where I explain `recoverWith` operation semantics.

Reply:
I have rewritten it entirely